File: src/components/info/augment/domina.ts
import Augment, { AugmentGroup } from "./_base";
import { EFFECTS as eff, TRIPLE_OFF_EFF } from "../_effect";

const GROUP = "DOMINA";

let augments: Array<Augment> = [];

augments.push({
    name: "ael domina",
    level: 0,
    effects: [
        { effect: eff.HP, amount: 5 },
        { effect: eff.PP, amount: 3 },
        ...TRIPLE_OFF_EFF.map((effect) => {
            return {
                effect,
                amount: 1.015,
            };
        }),
    ],
    battlepower: 8,
    group: GROUP,
});

augments.push({
    name: "ret domina",
    level: 0,
    effects: [
        { effect: eff.HP, amount: 15 },
        ...TRIPLE_OFF_EFF.map((effect) => {
            return {
                effect,
                amount: 1.015,
            };
        }),
    ],
    battlepower: 10,
    group: GROUP,
});

const DOMINA: AugmentGroup = {
    name: GROUP,
    augments,
};

export default DOMINA;

File: _data/augments/domina.py
from typing import List

from ._augment import Augment
from ._augment_groups import AugmentGroups
from effect import *
from effect import EffectTypes as ET
from util import many_effs_with_same_amount


GROUP = AugmentGroups.DOMINA
CONFLICT = (GROUP,)

augments: List[Augment] = []

# -------------------------------------------
# ael
augments.append(
    Augment(
        "ael domina",
        0,
        8,
        (
            Effect(ET.HP, 5),
            Effect(ET.PP, 3),
            *many_effs_with_same_amount(OFFENSIVE_POT, 1.015),
        ),
        GROUP,
        CONFLICT,
    )
)
# -------------------------------------------

# -------------------------------------------
# ret
augments.append(
    Augment(
        "ret domina",
        0,
        10,
        (
            Effect(ET.HP, 15),
            *many_effs_with_same_amount(OFFENSIVE_POT, 1.015),
        ),
        GROUP,
        CONFLICT,
    )
)
# -------------------------------------------
